src/cdk/v2/destinations/intercom/utils.js

```diff
@@ -0,0 +1,367 @@
+    lookupField = integrationsObj.lookup;
+  }
+  return lookupField;
+};
+
+/**
+ * Returns the value of name field
+ * @param {*} message
+ * @returns
+ */
+const getName = (message) => {
+  const name = message?.traits?.name || message?.context?.traits?.name;
+  if (name) return name;
+  const firstName = getFieldValueFromMessage(message, 'firstName');
+  const lastName = getFieldValueFromMessage(message, 'lastName');
+  if (firstName && lastName) {
+    return `${firstName} ${lastName}`;
+  }
+
+  if (firstName || lastName) {
+    return firstName || lastName;
+  }
+  return undefined;
+};
+
+/**
+ * Returns company payload
+ * @param {*} payload
+ * @returns
+ */
+const getCompaniesList = (payload) => {
+  const company = get(payload, 'custom_attributes.company');
+  if (!company) return undefined;
+  const companiesList = [];
+  if (company.name || company.id) {
+    const customAttributes = {};
+    Object.keys(company).forEach((key) => {
+      // If key is not in ReservedCompanyProperties
+      if (!ReservedCompanyProperties.includes(key)) {
+        const val = company[key];
+        if (val !== Object(val)) {
+          customAttributes[key] = val;
+        } else {
+          customAttributes[key] = JSON.stringify(val);
+        }
+      }
+    });
+
+    companiesList.push({
+      company_id: company.id || md5(company.name),
+      custom_attributes: removeUndefinedAndNullValues(customAttributes),
+      name: company.name,
+      industry: company.industry,
+    });
+  }
+  return companiesList;
+};
+
+/**
+ * Returns if email or userId is present in payload or not
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const checkIfEmailOrUserIdPresent = (message, Config) => {
+  const { context, anonymousId } = message;
+  let { userId } = message;
+  if (Config.sendAnonymousId && !userId) {
+    userId = anonymousId;
+  }
+  return !!(userId || context?.traits?.email);
+};
+
+/**
+ * Returns add user to company payload
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const attachUserAndCompany = (message, Config) => {
+  if (!checkIfEmailOrUserIdPresent(message, Config)) return undefined;
+  const email = message.context?.traits?.email;
+  const { userId, anonymousId, traits, groupId } = message;
+  const requestBody = {};
+  if (userId) {
+    requestBody.user_id = userId;
+  }
+  if (Config.sendAnonymousId && !userId) {
+    requestBody.user_id = anonymousId;
+  }
+  if (email) {
+    requestBody.email = email;
+  }
+  const companyObj = {
+    company_id: groupId,
+  };
+  if (traits?.name) {
+    companyObj.name = traits.name;
+  }
+  requestBody.companies = [companyObj];
+  const response = defaultRequestConfig();
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.endpoint = `${BASE_ENDPOINT}/users`;
+  response.headers = {
+    'Content-Type': JSON_MIME_TYPE,
+    Authorization: `Bearer ${Config.apiKey}`,
+    Accept: JSON_MIME_TYPE,
+    'Intercom-Version': '1.4',
```

Comment:
[Q] Is the hardcoding of version expected ? @mihir-4116

Reply:
In `getHeaders` we are using version check and putting a version value in `Intercom-version` header. Want to confirm if this is expected.

Reply:
yes, it is expected as it's old version code so Intercom-Version should be 1.4